Use UTC time for server stats history, localize times on the client

obradovichv opened the pull request:
Server-side stats history time sourcing changed from server time(time zone dependent) to UTC, aligning it with other server-side time values sourced from `time.time()`. This enables the same frame of reference for all clients by knowing it is UTC. Localization is now performed client-side on server-rendered dates and times.  

Localization of the `report.html` start and end datetimes intentionally formats the value according to the client's default locale to keep with the localization theme of this change.

The date strings are manually parsed instead of using `Date.parse()` as recommended by MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/parsed
`"It is not recommended to use Date.parse as until ES5, parsing of strings was entirely implementation dependent. There are still many differences in how different hosts parse date strings, therefore date strings should be manually parsed (a library can help if many different formats are to be accommodated)."`

Fixes #1835

cyberw commented:
I think supporting pre ES5 browsers doesnt really make it to our priority list (10 year old browsers are likely to have all kinds of other issues anyway), and I would prioritize keeping the code simple. Other than that it looks good!

obradovichv commented:
Agreed, the spread operator would likely invalidate any 10 year old browser anyway :laughing: 
Updated PR accordingly to use `Date.parse()` to parsing the datetime instead of manually parsing the value.

I have left `stats_data.html` manual parsing unchanged because it uses only the time component and would need an ISO8601-like date format to work with `Date.parse()`, perhaps the {current date} or {unix epoch} would be appropriate "fake" date components to pass into `Date.parse()` but it does not feel like a simplification of `"HH:MM".split(":")` by comparison. I can update to this if preferred.

If a future feature stores stat history with the full datetime for the time series, then the client-side localization can be updated to `Date.parse()` without any complicated date injections.

cyberw commented:
Thanks!

Do you by any chance have time to look into another thing related to graphing? I'd like to not have a line for response times at all until requests have actually been reported (the first part of the line in this screen shot)

![image](https://user-images.githubusercontent.com/1134328/129675145-d0795144-5b66-4f0f-98d0-4fdfca570481.png)

obradovichv commented:
Sure, I'll take a look.
Created new issue to track it: #1852

cyberw commented:
Awesome, thanks!
